message_processor.go

```diff
@@ -50,34 +85,50 @@ func consumeLog(lf *logPair) {
 			logrus.WithField("id", lf.info.ContainerID).WithError(err).Debug("Ignoring error")
 			dec = protoio.NewUint32DelimitedReader(lf.stream, binary.BigEndian, 1e6)
 		}
-		if sendMessage(lf.splunkl, &buf, lf.info.ContainerID) == false {
-			continue
-		}
-		if sendMessage(lf.jsonl, &buf, lf.info.ContainerID) == false {
-			continue
-		}
 
+		if shouldSendMessage(buf.Line) {
+			// Append to temp buffer
+			appendToTempBuffer(tmpBuf, &buf)
+			// Check for temp buffer timer expiration
+			diff := time.Now().Sub(tmpBuf.bufferTimer)
+			if diff > tempMsgBufferHoldDuration {
+				tmpBuf.bufferHoldDurationExpired = true
+			}
+
+			sendMessage(lf.splunkl, &buf, tmpBuf, lf.info.ContainerID)
+			sendMessage(lf.jsonl, &buf, tmpBuf, lf.info.ContainerID)
+			//temp buffer and values reset
+			if tmpBuf.bufferReset {
+				resetTempBufferValues(tmpBuf)
+			}
+		}
 		buf.Reset()
 	}
-
 }
 
 // send the log entry message to logger
-func sendMessage(l logger.Logger, buf *logdriver.LogEntry, containerid string) bool {
+func sendMessage(l logger.Logger, buf *logdriver.LogEntry, tBuffer *tmpBuffer, containerid string) bool {
```

Comment:
we don't need return bool any more

Reply:
I didn't know if I want to get rid of it. But, sure.